Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst

```diff
@@ -11,14 +11,136 @@ TYPO3 request object
 The TYPO3 request object is an implementation of the PSR-7 based
 `\Psr\Http\Message\ServerRequestInterface` containing TYPO3-specific attributes.
 
-The request object is passed to controllers. The attributes can be retrieved via
+..  seealso::
+    `PSR-7: HTTP message interfaces <https://www.php-fig.org/psr/psr-7/>`__
+
+.. contents::
+   :local:
+
+
+..  _getting-typo3-request-object:
+
+Getting the PSR-7 request object
+================================
+
+The PSR-7 based request object is available in most contexts. In some scenarios,
+like :ref:`PSR-15 middlewares <request-handling>` or backend module controllers,
+the PSR-7 base request object is given as argument to the called method.
+
+
+Extbase controller
+------------------
+
+..  versionadded:: 11.3
+
+The request object compatible with the PSR-7
+:php:`\Psr\Http\Message\ServerRequestInterface` is available in an
+:ref:`Extbase controller <extbase-action-controller>` via the class property
+:php:`$this->request`:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Controller/MyController.php
+
+    public function myAction(): ServerResponseInterface
```

Comment:
I would keep the class declaration in all of these examples, as the class that is being implented shows the context we are in without reading the text above. Also please keep the include statements (yes it is repetition about theServerResponseInterface but still usefull to new users )